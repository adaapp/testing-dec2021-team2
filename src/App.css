.App {
  text-align: center;
}

.App-logo {
  height: 300px;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  margin-top: 30px;
  min-height: 1vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  /* justify-content: center; */
  font-size: calc(10px + 2vmin);
  color: black;
}

.Preview {
  margin: auto;
  height: 100px;
  width: 30%;
  margin-bottom: 20px;
  border: 2px solid black;
  border-radius: 4px;
}

.Launch-Button {
  background-color: #282c34;
  border: 2px solid #282c34;
  border-radius: 4px;
  padding: 15px 40px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 20px;
  font-weight: bold;
  color: white;
  transition-duration: 0.4s;
}

.Launch-Button:hover {
  border: 2px solid #282c34;
  background-color: white;
  color: #282c34;
  cursor: pointer
}

input {
  margin: 10px;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
