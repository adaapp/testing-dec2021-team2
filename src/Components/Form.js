import React from 'react'
import { useState, useEffect } from 'react';
import '../App.css';
import convertRGB from './formFunctions';

// function that deals with all of this. 
export default function Form() {
	let [hex, setHex] = useState('');
	const [r, setR] = useState(0);
	const [g, setG] = useState(0);
	const [b, setB] = useState(0);

	function changeR(e) { 
		if (e.target.value > 255 || e.target.value < 0) {
			e.target.value = 255;
			setR(255);
		} else {
			setR(e.target.value)
		}
		
	};

	function changeG(e) { 
		if (e.target.value > 255 || e.target.value < 0) {
			e.target.value = 255;
			setG(255);
		} else {
			setG(e.target.value)
		}
	 };

	function changeB(e) { 
		if (e.target.value > 255 || e.target.value < 0) {
			e.target.value = 255; 
			setB(255);
		} else {
			setB(e.target.value)
		} 
	};

	useEffect(() => { setHex(convertRGB(r, g, b)); }, [r, g, b])

    return (
        <div className="App">
            <h1 data-testid="hex">{hex}</h1>
				<div data-testid="preview" className="Preview" style={{ backgroundColor: hex}}></div>
				<form>
					<label data-testid="r">
						R:
						<input type="number" name="r" max="255" min="0" onChange={changeR} />
					</label>
					<label data-testid="g">
						G:
						<input type="number" name="g" max="255" min="0" onChange={changeG} />
					</label>
					<label data-testid="b">
						B:
						<input type="number" name="b" max="255" min="0" onChange={changeB} />
					</label>
				</form>
        </div>
    )
}