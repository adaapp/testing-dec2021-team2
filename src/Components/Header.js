import React from 'react'
import '../App.css';
import logo from "../Images/Colourwheel.png"

function Launch() {
  window.open("https://github.com/adaapp/testing-dec2021-team2")
}

export default function Header() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Team 2: Colour Converter</h2>
          <form>
            <button className="Launch-Button" onClick={Launch}>See Project Files! ðŸš€</button>
          </form>
        </header>
      </div>
    )
}