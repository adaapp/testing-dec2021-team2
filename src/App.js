import './App.css';
import Header from './Components/Header'
import Form from './Components/Form'

function App() {
  return (
    <div className="Card">
      <Header />
      <Form />
    </div>
  );
}

export default App;
