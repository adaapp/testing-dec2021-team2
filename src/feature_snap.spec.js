import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Components/Header';
import Form from './Components/Form';
import App from './App';

it('checks that DOM for App is consistent', () => {
    const tree = renderer.create(<App/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('checks that DOM for Form is consistent', () => {
    const tree = renderer.create(<Form/>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('checks that DOM for Header is consistent', () => {
    const tree = renderer.create(<Header/>).toJSON();
    expect(tree).toMatchSnapshot();
});

