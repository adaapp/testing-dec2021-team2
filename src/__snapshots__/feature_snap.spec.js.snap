// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks that DOM for App is consistent 1`] = `
<div
  className="Card"
>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="Colourwheel.png"
      />
      <h2>
        Team 2: Colour Converter
      </h2>
      <form>
        <button
          className="Launch-Button"
          onClick={[Function]}
        >
          See Project Files! ðŸš€
        </button>
      </form>
    </header>
  </div>
  <div
    className="App"
  >
    <label>
      Hex:
      <input
        data-testid="hex"
        name="r"
        onChange={[Function]}
        value=""
      />
    </label>
    <div
      className="Preview"
      data-testid="preview"
      style={
        Object {
          "backgroundColor": "",
        }
      }
    />
    <form>
      <label
        data-testid="r"
      >
        R:
        <input
          data-testid="rInput"
          name="r"
          onBlur={[Function]}
          onChange={[Function]}
          style={
            Object {
              "width": "100px",
            }
          }
          value=""
        />
      </label>
      <label
        data-testid="g"
      >
        G:
        <input
          data-testid="gInput"
          name="g"
          onBlur={[Function]}
          onChange={[Function]}
          style={
            Object {
              "width": "100px",
            }
          }
          value=""
        />
      </label>
      <label
        data-testid="b"
      >
        B:
        <input
          data-testid="bInput"
          name="b"
          onBlur={[Function]}
          onChange={[Function]}
          style={
            Object {
              "width": "100px",
            }
          }
          value=""
        />
      </label>
    </form>
  </div>
</div>
`;

exports[`checks that DOM for Form is consistent 1`] = `
<div
  className="App"
>
  <label>
    Hex:
    <input
      data-testid="hex"
      name="r"
      onChange={[Function]}
      value=""
    />
  </label>
  <div
    className="Preview"
    data-testid="preview"
    style={
      Object {
        "backgroundColor": "",
      }
    }
  />
  <form>
    <label
      data-testid="r"
    >
      R:
      <input
        data-testid="rInput"
        name="r"
        onBlur={[Function]}
        onChange={[Function]}
        style={
          Object {
            "width": "100px",
          }
        }
        value=""
      />
    </label>
    <label
      data-testid="g"
    >
      G:
      <input
        data-testid="gInput"
        name="g"
        onBlur={[Function]}
        onChange={[Function]}
        style={
          Object {
            "width": "100px",
          }
        }
        value=""
      />
    </label>
    <label
      data-testid="b"
    >
      B:
      <input
        data-testid="bInput"
        name="b"
        onBlur={[Function]}
        onChange={[Function]}
        style={
          Object {
            "width": "100px",
          }
        }
        value=""
      />
    </label>
  </form>
</div>
`;

exports[`checks that DOM for Header is consistent 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="Colourwheel.png"
    />
    <h2>
      Team 2: Colour Converter
    </h2>
    <form>
      <button
        className="Launch-Button"
        onClick={[Function]}
      >
        See Project Files! ðŸš€
      </button>
    </form>
  </header>
</div>
`;
